src/taipy/core/data/generic.py

```diff
@@ -101,15 +111,19 @@ def storage_type(cls) -> str:
         return cls.__STORAGE_TYPE
 
     def _read(self):
-        if read_fct := self.properties[self._REQUIRED_READ_FUNCTION_PROPERTY]:
-            if self._READ_FUNCTION_PARAMS_PROPERTY in self.properties.keys():
-                return read_fct(*self.properties[self._READ_FUNCTION_PARAMS_PROPERTY])
+        if read_fct := self.properties[self._OPTIONAL_READ_FUNCTION_PROPERTY]:
+            if read_fct_args := self.properties.get(self._READ_FUNCTION_ARGS_PROPERTY, None):
+                if not isinstance(read_fct_args, list):
+                    return read_fct(*[read_fct_args])
+                return read_fct(*read_fct_args)
             return read_fct()
         raise MissingReadFunction(f"The read function is not defined in data node config {self.config_id}.")
 
     def _write(self, data: Any):
-        if write_fct := self.properties[self._REQUIRED_WRITE_FUNCTION_PROPERTY]:
-            if self._WRITE_FUNCTION_PARAMS_PROPERTY in self.properties.keys():
-                return write_fct(data, *self.properties[self._WRITE_FUNCTION_PARAMS_PROPERTY])
+        if write_fct := self.properties[self._OPTIONAL_WRITE_FUNCTION_PROPERTY]:
+            if write_fct_args := self.properties.get(self._WRITE_FUNCTION_ARGS_PROPERTY, None):
+                if not isinstance(write_fct_args, list):
+                    return write_fct(data, *[write_fct_args])
+                return write_fct(data, *write_fct_args)
```

Comment:
yep still do, I'm assuming you're talking about line 125 and 126, we require args to be list and we enforce it in dataconfigchecker, but just in case somehow some ways, the generic dn was created without the use of config, it will still work. It's another layer to avoid system exit error :D

Reply:
FYI, it is important to keep the if condition since the data node entity may have been modified after its creation. That means the entity could be wrong even if the config is correct.